src/platforms/hosted/stlinkv2.c

```diff
@@ -676,20 +678,25 @@ static void stlink_mem_read(adiv5_access_port_s *ap, void *dest, uint32_t src, s
 {
 	if (len == 0)
 		return;
-	if (len > stlink.block_size) {
-		DEBUG_WARN("Too large!\n");
-		return;
-	}
 	if (!stlink_ensure_ap(ap->apsel))
 		raise_exception(EXCEPTION_ERROR, "ST-Link AP selection error");
 
 	uint8_t type;
-	if ((src & 1U) || (len & 1U))
+	uint16_t block_size;
+	if ((src & 1U) || (len & 1U)) {
 		type = STLINK_DEBUG_READMEM_8BIT;
-	else if ((src & 3U) || (len & 3U))
+		block_size = stlink.block_size;
+	} else if ((src & 3U) || (len & 3U)) {
 		type = STLINK_DEBUG_APIV2_READMEM_16BIT;
-	else
+		block_size = 4096;
```

Comment:
This and the one below are missing their `U`'s. It might be helpful to document why this is OK and where this block size comes from - no objections otherwise.

Reply:
Can add unsigned qualifiers. For size, different impls actually lean towards 6144 bytes per read, which sounds realistic for the 20/64 f103c8, but earlier firmware might only do 4096 (?). I don't have a good testbench for this, and AHB-AP won't autoincrement past 4096 byte boundaries anyways (for CM3/CM4), so this is a sensible compromise between not having any access at all and surfing the dangerous edge. *For JLink fw which has 3 buffers of 2 KiB for TMS, TDO, TDI this also lines up somewhat.* Anecdotally, writes are streamed for up to (65536-4) bytes, but don't @ me on this -- see probe-rs PR377, `src/probe/stlink/mod.rs` for example.

Reply:
Sure, having that as a comment near the top of the file or in the ST-Link v2 protocol header and using a define for it is probably favourite then. That puts the reasoning somewhere easily found in the future that helps describe why the magic number and not, eg, the probe-rs choice, etc.. The value picked is fine, we just want to capture the reasoning somehow.

Reply:
Extracted macro and moved to protocol header. Failing to find any public docs I bruteforced the read length for a bit, ending up on 6144.